odoo/fields.py

```diff
@@ -2120,14 +2120,37 @@ class Selection(Field):
             > result = [('a', 'A'), ('c', 'C'), ('b', 'B')]
     :type selection_add: list(tuple(str,str))
 
+    :param ondelete: provides a fallback mechanism for any overridden
+        field with a selection_add. It is a dict that maps every option
+        from the selection_add to a fallback action.
+
+        This fallback action will be applied to all records whose
+        selection_add option maps to it.
+
+        The actions can be any of the following:
+            - 'null' -- the default, all records with this option
+              will have their selection value set to False.
+            - 'cascade' -- all records with this option will be
+              deleted along with the option itself.
+            - 'default' -- all records with this option will be
+              set to the default of the field definition
+            - 'set <value>' -- all records with this option will be
+              set to the specified <value>, this <value> needs to be
+              a valid selection option.
+            - <callable> -- a callable whose first and only argument will be
+              the set of records containing the specified Selection option,
+              for custom processing
+
     The attribute ``selection`` is mandatory except in the case of
     ``related`` or extended fields.
     """
     type = 'selection'
     column_type = ('varchar', pg_varchar())
     _slots = {
-        'selection': None,              # [(value, string), ...], function or method name
-        'validate': True,               # whether validating upon write
+        'selection': None,          # [(value, string), ...], function or method name
+        'validate': True,           # whether validating upon write
+        'ondelete': None,           # {value: policy},There must be one ondelete action per Selection option what to do when the value is deleted
+        'extended': False,          # whether this field is extended by another with selection_add
```

Comment:
What is the purpose of `extended`?

Reply:
it was for determining whether the current field defined a `selection_add` and then checking it inside `_setup_regular_base`, but since it's all done inside `_setup_attrs` now, there's no need